src/main/java/com/streamr/client/utils/Web3jUtils.java

```diff
@@ -0,0 +1,197 @@
+        Condition txfinish = new Condition(){
+            @Override
+            public Object check() throws Exception {
+                Optional<TransactionReceipt> tr = web3j.ethGetTransactionReceipt(txhash).send().getTransactionReceipt();
+                if(!tr.isPresent())
+                    return null;
+                return tr.get().isStatusOK();
+            }
+        };
+        return (Boolean) waitForCondition(txfinish, sleeptime, timeout);
+    }
+
+    /**
+     *
+     * @param initialBalance
+     * @param tokenAddress
+     * @param balanceAddress
+     * @param connector
+     * @param sleeptime
+     * @param timeout
+     * @return new balance, or null if balance didnt change in waiting period
+     * @throws Exception
+     */
+
+    public static BigInteger waitForErc20BalanceChange(BigInteger initialBalance, String tokenAddress, String balanceAddress, Web3j connector, long sleeptime, long timeout) throws Exception {
+        Object o = waitForCondition(new Erc20BalanceChanged(initialBalance, tokenAddress, balanceAddress, connector), sleeptime, timeout);
+        return o == null ? null : (BigInteger) o;
+    }
+
+    public static BigInteger callUintGetterFunction(String contract, String functionName, Web3j connection) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException, IOException {
+        Function fn = FunctionEncoder.makeFunction(functionName,
+                Collections.<String>emptyList(),
+                Collections.emptyList(),
+                Arrays.<String>asList("uint256")
+        );
+        List<Type> ret = callFunction(contract, fn, connection);
+        return (BigInteger) ret.iterator().next().getValue();
+    }
+
+    public static BigInteger erc20Balance(String contract, String balanceAddress, Web3j connection) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException, IOException {
+        Function fn = FunctionEncoder.makeFunction("balanceOf",
+                Arrays.<String>asList("address"),
+                Arrays.asList(balanceAddress),
+                Arrays.<String>asList("uint256")
+        );
+        List<Type> ret = callFunction(contract, fn, connection);
+        return (BigInteger) ret.iterator().next().getValue();
+    }
+
+    public static String callAddressGetterFunction(String contract, String functionName, Web3j connection) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException, IOException {
+        Function fn = FunctionEncoder.makeFunction(functionName,
+                Collections.<String>emptyList(),
+                Collections.emptyList(),
+                Arrays.<String>asList("address")
+        );
+        List<Type> ret = callFunction(contract, fn, connection);
+        return (String) ret.iterator().next().getValue();
+    }
```

Comment:
Unused. Remove.

Reply:
I dont see the harm in having this generic address static getter. @hpihkala ?

Reply:
I removed